src/components/MovieListComponent.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { useSearchParams, Outlet } from "react-router-dom";
+
+import Fetch from '../services/Fetch.js'
+import MoviesList from './MoviesList.js'
+import GenreSelectComponent from './GenreSelectComponent.js'
+import SortControl from './SortControl.js'
+
+function MovieListComponent() {
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [searchQuery, setSearchQuery] = useState('');
+    const [searchResults, setSearchResults] = useState(null);
+    const [sortCriterion, setSortCriterion] = useState('');
+    const genres = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];
+    const [activeGenre, setActiveGenre] = useState('');
+
+
+
+const initializeStateFromSearchParams = () => {
+    const query = searchParams.get('searchQuery');
+    const genre = searchParams.get('genre');
+    const sortCriterion = searchParams.get('sortBy');
+
+    if (query) {
+      setSearchQuery(query);
+    }
+
+    if (genre) {
+      setActiveGenre(genre);
+    }
+
+    if (sortCriterion) {
+      setSortCriterion(sortCriterion);
+    }
+
+  };
+
+    const handleSearchSubmit = (searchQueryFromInput) => {
+        setSearchParams({ searchQuery: searchQueryFromInput});
+        setSearchQuery(searchQueryFromInput);
+        setActiveGenre('All');
+      };
+    
+
+    useEffect(() => {
+        initializeStateFromSearchParams();
```

Comment:
You don't need this effect. You can use searchParams as default values in useState.